spec/cc/engine/analyzers/go/main_spec.rb

```diff
@@ -0,0 +1,180 @@
+require "spec_helper"
+require "cc/engine/analyzers/go/main"
+require 'cc/engine/analyzers/reporter'
+require "cc/engine/analyzers/engine_config"
+
+module CC::Engine::Analyzers
+  RSpec.describe Go::Main, in_tmpdir: true do
+    include AnalyzerSpecHelpers
+
+    describe "#run" do
+      it "prints an issue for identical code" do
+        create_source_file("foo.go", <<-EOGO)
+          package main
+
+          import "fmt"
+
+          func main() {
+              some_string()
+          }
+
+          func some_string() {
+              fmt.Println("this is a string")
+              fmt.Println("this is a string")
+          }
+
+          func some_string() {
+              fmt.Println("this is a string")
+              fmt.Println("this is a string")
+          }
+
+          func some_string() {
+              fmt.Println("this is a string")
+              fmt.Println("this is a string")
+          }
+        EOGO
+
+        issues = run_engine(engine_conf).strip.split("\0")
+        result = issues.first.strip
+        json = JSON.parse(result)
+
+        expect(json["type"]).to eq("issue")
+        expect(json["check_name"]).to eq("identical-code")
+        expect(json["description"]).to eq("Identical blocks of code found in 3 locations. Consider refactoring.")
+        expect(json["categories"]).to eq(["Duplication"])
+        expect(json["location"]).to eq({
+          "path" => "foo.go",
+          "lines" => { "begin" => 9, "end" => 12 },
+        })
+        expect(json["remediation_points"]).to eq(380_000)
+        expect(json["other_locations"]).to eq([
+          {"path" => "foo.go", "lines" => { "begin" => 14, "end" => 17} },
+          {"path" => "foo.go", "lines" => { "begin" => 19, "end" => 22} },
+        ])
+        expect(json["content"]["body"]).to match(/This issue has a mass of 32/)
+        expect(json["fingerprint"]).to eq("729e7221b4530916ed63cfb6f4a3fe90")
+        expect(json["severity"]).to eq(CC::Engine::Analyzers::Base::MINOR)
+      end
+
+      it "prints an issue for similar code" do
+        create_source_file("foo.go", <<-EOGO)
+          package main
+
+          import "fmt"
+
+          func add(x int, y int) int {
+          	return x + y
+            fmt.Println("Add some stuff!")
+          }
+
+          func add(x int, y int) int {
+            return x + y
+            fmt.Println("Add all the stuff!")
+          }
+
+          func main() {
+          	fmt.Println(add(42, 13))
```

Comment:
An indentation quibble: there might be a mix of tabs and spaces here and line 66.